ldap.sgnl.yaml

```diff
@@ -0,0 +1,151 @@
+#   "entityConfig": {
+#     "<EntityName-1>": {
+#       "query": "(&(objectCategory=user)(objectClass=user)(distinguishedName=*))"
+#     },
+#     "<EntityName-2>": {
+#       "query": "(&(objectCategory=group)(objectClass=group)(distinguishedName=*))"
+#     },
+#     "<EntityName-3>": {
+#       "memberOf": <EntityName-2>,
+#       "query": "(&(memberOf={{CollectionId}})(objectCategory=user)(objectClass=user))",
+#       "memberUniqueIdAttribute": "memberObjectGUID",
+#       "memberOfUniqueIdAttribute": "groupObjectGUID"
+#     }
+#   },
+# }
+
+# Adapter Config Field Info:
+# [Required] baseDN (String): Specifies the Base Distinguished Name (DN) for the LDAP directory.
+
+# [Required] certificateChain (String): Contains the Base64 encoded certificate chain for secure communication with the LDAP server.
+
+# [Optional] entityConfig (Object): Configurations for different types of entities within the LDAP directory.
+
+# For more information on LDAP queries: https://ldap.com/ldap-filters/
+
+# [Optional] entityConfig.<EntityName> (Object): Configuration for a specific type of entity.
+# [Optional] entityConfig.<EntityName>.query (String): Specifies the LDAP query for retrieving entities of this type.
+
+# [Optional] entityConfig.<EntityName>.memberOf (String): Specifies the group or entity to which the members belong.
+# [Optional] entityConfig.<EntityName>.query (String): Specifies the LDAP query for retrieving entities of this type.
+# [Optional] entityConfig.<EntityName>.memberUniqueIdAttribute (String): Specifies the attribute used as a unique identifier for members.
+# [Optional] entityConfig.<EntityName>.memberOfUniqueIdAttribute (String): Specifies the attribute used as a unique identifier for the group or parent entity.
+
+
+adapterConfig: "ewogICAgImJhc2VETiI6ICJ7e0lucHV0IFJlcXVpcmVkOiBCYXNlRE59fSIsCiAgICAiY2VydGlmaWNhdGVDaGFpbiI6ICJ7e0lucHV0IFJlcXVpcmVkOiBJZiBMREFQUyBpcyBlbmFibGVkOiBiYXNlNjQgZW5jb2RlZCBjZXJ0aWZpY2F0ZUNoYWlufX0iCn0="
+# All auth mechanisms specified below must be supported by the specified Adapter.
+auth:
+  - basic:
+      username: "{{Input Required: Username}}"
+      password: "{{Input Required: Password}}"
+entities:
+  # Reference for Active Directory Schema
+  # See: https://learn.microsoft.com/en-us/windows/win32/adschema/active-directory-schema
+  <EntityName-1>:
+    displayName: <EntityName-1>
+    externalId: <EntityName-1>
+    description: An entity representing <EntityName-1> in an LDAP
+    pageSize: 100
+    pagesOrderedById: false
+    attributes:
+      - name: objectGUID
+        externalId: objectGUID
+        description: The objectGUID of the User
+        type: String
+        indexed: true
+        uniqueId: true
+        attributeAlias: userGUID
+      - name: dn
+        externalId: dn
+        description: The Distinguished Name (DN) of the user
+        type: String
+      - name: objectClass
+        externalId: objectClass
+        type: String
+        list: true    
```

Comment:
For entities, let's have a simple example of a couple of entities similar to user, group and groupmember i.e. with atleast one collection entity from which we derive membership data.

As this an example, to facilitate easy plug and play, we build the catalog around a free live server available online (https://www.forumsys.com/2022/05/10/online-ldap-test-server/)

WDYT @Mohitp98, @marc-sgnl

Reply:
After chatting with @marc-sgnl offline, we've decided that we don't need a sample template for LDAP in the catalogs. Will be closing this PR.